domain/src/main/java/org/oppia/domain/onboarding/OnboardingFlowCache.kt

```diff
@@ -0,0 +1,33 @@
+package org.oppia.domain.onboarding
+
+import org.oppia.app.model.OnboardingFlow
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.util.logging.Logger
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** This class creates a PersistentCacheStore for onboarding flow. */
+@Singleton
+class OnboardingFlowCache @Inject constructor(
+  cacheStoreFactory: PersistentCacheStore.Factory,
+  private val logger: Logger
+) {
+  private val onboardingFlowStore = cacheStoreFactory.create("on_boarding_flow", OnboardingFlow.getDefaultInstance())
```

Comment:
Hmm, not sure I’m keen on sharing the whole store like this. 
Might we instead just share the name of it so that the test helper can access that to access the file directly for deletion?

Reply:
@BenHenning Meaning we should just share `on_boarding_flow` name and create separate cachestores in each file? Also if so, then creating it as `const val` will work?

Reply:
Meaning we can go back to previous implementation in which the cacheStoreFactory was in OnboardingFlowController and in that we can make `on_boarding_flow ` as `const val` which will be used in helper class to create another cacheStorFactory similar to OnboardingFlowController

Is this correct?

Reply:
Yes to having two cache store objects (though I’m not sure you need two, can’t you just delete the file directly via the test helper?). Regarding how to share the name, I think it would be a bit cleaner to share it via dagger. The name is an implementation detail if it’s part of the class, and putting a dependency directly on that detail doesn’t sound great. If you use a dagger constant, it becomes its own entity that can be shared everywhere that needs it (eg via injection). It’s still not a great long term solution (we ought to have a way to just clear all files, which I suppose you could do via a test helper as well), but it at least keeps the dependencies a bit cleaner.

Reply:
Okay, sounds clearer now. I will update the code accordingly.
@BenHenning Thanks.

Reply:
As suggested by Ben in chats, just recreating the root component was sufficient to simulate the app from scratch, so that splash screen can open onboarding flow activity after recreating the root component.